python/tvm/topi/hexagon/tensor_intrin.py

```diff
@@ -73,6 +73,58 @@ def _q_multiply_shift_hexagon(op):
 )
 
 
+def _q_multiply_shift_per_axis_hexagon(op):
+    """
+    Implementation of q_multiply_shift_per_axis through hexagon intrinsics vmpyewuh and vmpyowh when
+    q == 31.
+
+    Please note that this is introducing a small round-up error for some corner cases when we do
+    right shift. This is because we are rounding twice instead than only once. I.e.:
+
+        * original q_multiply_shift: round(x*y*2^-s)
+        * hexagon q_multiply_shift: round(round(x*y)*2^-s)
```

Comment:
cc @kparzysz-quic @jverma-quic on this HVX implementation.

Reply:
I will add test to demonstrate issue with accuracy.

Reply:
I have fixed issue with accuracy drop.
For the case when we need both right and left shifts at the same time I use "old" approach and lower this oper to the sequence left_shift/multipy/add/right_shift (64bit arithmetic). Right now I have no idea how to implement this case through vector HVX instructions without accuracy drop.